lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -345,16 +345,41 @@ def scoped_privileges(user_or_attribute, object_or_attribute, options = {})
       end
     end
 
+    ## Optimized version of PolicyMachine#scoped_privileges
+    # Returns all objects the user has the given operation on
+    # TODO: Support multiple policy classes here
+    def accessible_objects(user_or_attribute, operation, options = {})
+      operation = class_for_type('operation').find_by_unique_identifier!(operation.to_s) unless operation.is_a?(class_for_type('operation'))
+      permitting_oas = PolicyElement.where(id: operation.policy_element_associations.where(
```

Comment:
```sql
SELECT "policy_elements".* FROM "policy_elements" WHERE "policy_elements"."id" IN (SELECT "policy_element_associations"."object_attribute_id" FROM "policy_element_associations" INNER JOIN "operations_policy_element_associations" ON "policy_element_associations"."id" = "operations_policy_element_associations"."policy_element_association_id" WHERE "operations_policy_element_associations"."operation_id" = 4 AND "policy_element_associations"."user_attribute_id" IN (7, 6))
```

So two calls, I think, one for `.descendants` and one for the rest. I think that's because `|` on a chain isn't working right anymore, that's a problem with at least one other method in this file too.